### PR TITLE
API: change default behavior in `Table.pformat` and `Column.pformat` and add (pending) deprecation to `Table.pformat_all`

### LINKED ISSUE #15828: Consider changing the default behavior of `Table.pformat()` to be independent of the terminal
### What is the problem this feature will solve?

Currently the `Table.pformat()` default is to respect the size of the interactive terminal when generating lines of output representing the formatted table. This matches what would be seen with the corresponding `Table.pprint()` call (more-or-less).

The problem is that in most cases `pformat()` is used in a context where an interactive terminal is not relevant or does not exist (in which case 24x80 is assumed). This leads to unexpected results or the need for workarounds. A lot of test code uses `pformat_all()` as a work-around. 

Issue #14010 is a direct result of this unexpected behavior because `str(table)` also respects the terminal size.

### Describe the desired outcome

By default `Table.pformat()` would be equivalent to `Table.pformat_all()`, i.e. use `max_lines=-1` and `max_width=-1`. This would make `pformat()` independent of the terminal and lead to fewer suprises.

### Additional context

The big concern is whether this change will cause unexpected code breakage. This would need to wait until 7.0 but if merged into `main` there would be plenty of time for testing.

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
Fixes #15828, replaces #17179

EDIT: Closes #17179

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: docs/changes/table/17184.api.rst
Comment (taldcroft) [inline docs/changes/table/17184.api.rst:3]:
```suggestion
The default behavior of ``Table.pformat`` was changed to include all rows and columns
instead of truncating the outputs to fit the current terminal.  The new default
keyword arguments ``max_width=-1`` and ``max_lines=-1`` now match those in
``Table.pformat_all``. Since the ``Table.pformat_all`` method is now redundant, it is
pending deprecation. Similarly, the default behavior of ``Column.pformat`` was changed
to include all rows instead of truncating the outputs to fit the current terminal.
```

**Review by taldcroft — COMMENTED** [review]
Looks great, thanks! I just have some word-smithing suggestions. I think this should also get a What's new entry.

File: astropy/table/table.py
Comment (neutrinoceros) [reply astropy/table/table.py:2010]: the phrasing I used is borrowed from the only other pending deprecation (`Table.show_in_browser`). I'm also not sure it's worth doing a *pending* deprecation if we're 100% sure that it will be deprecated; why not just deprecate it now then ?

File: astropy/table/column.py
Comment (taldcroft) [inline astropy/table/column.py:909]:
```suggestion
        If ``max_lines=None`` (default) then the height of the
```

File: astropy/table/column.py
Comment (taldcroft) [inline astropy/table/column.py:914]: Revert.
```suggestion
        there is no line limit applied.
```

File: docs/whatsnew/7.0.rst
Comment (taldcroft) [inline docs/whatsnew/7.0.rst:66]:
```suggestion
``Table.pformat`` is now independent of terminal dimensions
```

**Review by taldcroft — COMMENTED** [review]
Just some little documentation nits now, otherwise all good!

**Comment by taldcroft** [comment]
Needs a rebase. :-(

**Comment by neutrinoceros** [comment]
suggestions integrated and branch rebased !

**Comment by taldcroft** [comment]
Sorry, this needs a rebase as well. But now I have auto-merge enabled so it should go through!

**Comment by pllim** [comment]
@taldcroft , you need to approve for auto merge to happen. 😉

**Comment by neutrinoceros** [comment]
Something seems stuck, all required checks are green but auto-merge still didn't happen. I'll try to disable it an merge manually.

**Comment by pllim** [comment]
Well, if you keep the auto-merge, it would have merged as soon as approval is granted.

**Review by taldcroft — APPROVED** [review]
Looks good!

**Pull request merged**
